File: app/src/main/java/dev/chu/architecture/java/p87/MainFragmentSubcomponent.java
package dev.chu.architecture.java.p87;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;
import dev.chu.architecture.etc.FragmentScope;

@FragmentScope
@Subcomponent(modules = MainFragmentModule.class)
public interface MainFragmentSubcomponent extends AndroidInjector<MainFragment> {

    @Subcomponent.Factory
    interface Factory extends AndroidInjector.Factory<MainFragment> {

    }

}

File: app/src/main/java/dev/chu/architecture/kotlin/p87/MainFragmentSubcomponent.kt
package dev.chu.architecture.kotlin.p87

import dagger.Subcomponent
import dagger.android.AndroidInjector
import dev.chu.architecture.etc.FragmentScope

// 보일러 플레이트 제거

@Subcomponent(modules = [MainFragmentModule::class])
@FragmentScope
interface MainFragmentSubcomponent: AndroidInjector<MainFragment> {

    @Subcomponent.Factory
    interface Factory: AndroidInjector.Factory<MainFragment>

}
